src/combinatorics/quiver/iso.lean

```diff
@@ -0,0 +1,187 @@
+
+instance : inhabited (iso U U) := ⟨iso.refl U⟩
+
+/-- Swapping `to_prefunctor` and `inv_prefunctor` inverts an isomorphism. -/
+@[simps] def iso.symm (φ : iso U V) : iso V U :=
+⟨φ.inv_prefunctor, φ.to_prefunctor, φ.right_inv, φ.left_inv⟩
+
+/-- Composing the components of two isomorphisms. -/
+@[simps] def iso.trans (φ : iso U V) (ψ : iso V W) : iso U W :=
+{ to_prefunctor := φ.to_prefunctor ⋙q ψ.to_prefunctor,
+  inv_prefunctor := ψ.inv_prefunctor ⋙q φ.inv_prefunctor,
+  left_inv := by
+  { rw [←prefunctor.comp_assoc, prefunctor.comp_assoc φ.to_prefunctor,
+        ψ.left_inv, prefunctor.comp_id, φ.left_inv], },
+  right_inv := by
+  { rw [←prefunctor.comp_assoc, prefunctor.comp_assoc ψ.inv_prefunctor,
+        φ.right_inv, prefunctor.comp_id, ψ.right_inv], }, }
+
+/--
+The equivalence on vertices induced by an isomorphism.
+-/
+@[simps] def iso.to_equiv (φ : iso U V) : U ≃ V :=
+{ to_fun := φ.to_prefunctor.obj,
+  inv_fun := φ.inv_prefunctor.obj,
+  left_inv := λ x, congr_arg (λ (F : U ⥤q U), F.obj x) φ.left_inv,
+  right_inv := λ x, congr_arg (λ (F : V ⥤q V), F.obj x) φ.right_inv }
+
+lemma iso.inv_obj_obj_eq (φ : iso U V) (X : U) : φ.inv_prefunctor.obj (φ.to_prefunctor.obj X) = X :=
+φ.to_equiv.left_inv X
+
+lemma iso.obj_inv_obj_eq (φ : iso U V) (X : V) : φ.to_prefunctor.obj (φ.inv_prefunctor.obj X) = X :=
+φ.to_equiv.right_inv X
+
+lemma iso.to_obj_injective (φ : iso U V) : φ.to_prefunctor.obj.injective :=
+φ.to_equiv.injective
+
+lemma iso.inv_obj_injective (φ : iso U V) : φ.inv_prefunctor.obj.injective :=
+φ.symm.to_equiv.injective
+
+/--
+The equivalence on arrows `X ⟶ Y ≃ (φ.obj X ⟶ φ.obj Y)` induced by the isomorphism `φ`,
+The forward map is `φ.to_prefunctor.map`, but the backward map is the composite of
+* `φ.inv_prefunctor.map : φ.obj X ⟶ φ.obj Y → φ.symm.obj (φ.obj X) ⟶ φ.symm.obj (φ.obj Y)`, and
+* `hom.equiv_cast _ _ : φ.symm.obj (φ.obj X) ⟶ φ.symm.obj (φ.obj Y) → X ⟶ Y`.
+-/
+@[simps] def iso.to_equiv_hom (φ : iso U V) {X Y : U} : (X ⟶ Y) ≃ (φ.obj X ⟶ φ.obj Y) :=
+{ to_fun := φ.to_prefunctor.map,
+  inv_fun := hom.equiv_cast (φ.to_equiv.left_inv X) (φ.to_equiv.left_inv Y) ∘ φ.inv_prefunctor.map,
+  left_inv := λ e, by
+    begin
+      nth_rewrite_rhs 0 ←((prefunctor.id_map _ _ _ e).rec_on $
+                           prefunctor.map_cast_eq_of_eq φ.left_inv e),
+      simp only [function.comp_app, prefunctor.comp_map, hom.equiv_cast_apply],
+      apply hom.cast_irrelevant,
+    end,
+  right_inv := λ e, by
+    begin
+      nth_rewrite_rhs 0 ←((prefunctor.id_map _ _ _ e).rec_on $
+                           prefunctor.map_cast_eq_of_eq φ.right_inv e),
+      simp only [prefunctor.map_cast, function.comp_app, prefunctor.comp_map, hom.equiv_cast_apply],
+      apply hom.cast_irrelevant,
+    end }
+
+/--
+The equivalence on arrows `X ⟶ Y ≃ φ.symm.obj X ⟶ φ.symm.obj Y` induced by `φ`.
+-/
+@[simp] def iso.to_equiv_hom' (φ : iso U V) {X Y : V} :
+  (X ⟶ Y) ≃ (φ.symm.obj X ⟶ φ.symm.obj Y) := φ.symm.to_equiv_hom
```

Comment:
I would drop this definition. It's syntactically less general than the previous.